shared/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -155,6 +155,82 @@ namespace datadog::shared::nativeloader
             }
         }
 
+        //
+        // Get Profiler interface ICorProfilerInfo4
+        //
+        ICorProfilerInfo4* info4 = nullptr;
+        HRESULT hr = pICorProfilerInfoUnk->QueryInterface(__uuidof(ICorProfilerInfo4), (void**) &info4);
+        if (FAILED(hr))
+        {
+            Log::Warn("CorProfiler::Initialize: Failed to attach profiler, interface ICorProfilerInfo4 not found.");
+            return E_FAIL;
+        }
+        const auto runtimeInformation = GetRuntimeVersion(info4);
+
+        //
+        // Check if we're running in Single Step, and if so, whether we should bail out
+        //
+        // This variable is non-empty when we're in single step
+        const auto isSingleStepVariable = GetEnvironmentValue(EnvironmentVariables::SingleStepInstrumentationEnabled);
+        if(!isSingleStepVariable.empty())
+        {
+            // We're doing single-step instrumentation, check if we're in an EOL environment
+            IUnknown* tstVerProfilerInfo;
+            const char* unsupportedFramework = nullptr;
+            if (runtimeInformation.is_core())
+            {
+                // For .NET Core, we require .NET Core 3.1 or higher
+                // We could use runtime_information, but I don't know how reliable the values we get from that are?
+                if (S_OK == pICorProfilerInfoUnk->QueryInterface(__uuidof(ICorProfilerInfo11), (void**) &tstVerProfilerInfo))
+                {
+                    // supported
+                    tstVerProfilerInfo->Release();
+                }
+                else
+                {
+                    unsupportedFramework = ".NET Core 3.0";
+                }
+            }
+            else
+            {
+                // For .NET Framework, we require .NET Framework 4.6.1 or higher
```

Comment:
Don't we do this already somewhere else? (not related to SSI)

Reply:
We do this in the native tracer, but not in the native loader currently, but we want to bail out earlier here